src/devtools/client/webconsole/components/Search/useConsoleSearch.test.tsx

```diff
@@ -35,107 +45,56 @@ describe("useConsoleSearch", () => {
-      repeatId: null,
-      source: "",
-      stacktrace: [],
-      type,
-      lastExecutionPoint: {
-        point: "1",
-        time: 1,
-        messageCount: 0,
-      },
-    };
-  }
-
-  function createValueFront(value: any): ValueFront {
-    return new ValueFront(null, { value });
-  }
-
-  async function createStore(messages: Message[] = []): Promise<UIStore> {
-    const messageIDs = messages.map(message => message.id);
-    const idToMessageMap = messages.reduce((map, message) => {
-      return {
-        ...map,
-        [message.id]: message,
-      };
-    }, {});
-
-    const defaultState = (await createTestStore()).getState();
-
-    return createTestStore({
-      ...defaultState,
-      app: {
-        ...defaultState.app,
-        loadedRegions: {
-          indexed: [],
-          loaded: [
-            {
-              begin: {
-                point: "0",
-                time: 0,
-              },
-              end: {
-                point: "1000",
-                time: 1000,
-              },
-            },
-          ],
-          loading: [],
-        },
-      },
-      messages: {
-        ...defaultState.messages,
-        messages: {
-          ids: messageIDs,
-          entities: idToMessageMap,
-        },
-        visibleMessages: messageIDs,
-      },
-    });
-  }
-
-  async function render(store?: UIStore) {
-    if (!store) {
-      store = await createStore([
-        createMessage("logPoint", "Plain string message"),
-        createMessage("logPoint", "", [
-          createValueFront(123),
-          createValueFront("string value"),
-          createValueFront(true),
-        ]),
-        createMessage("logPoint", "Another string message"),
-        createMessage("logPoint", "", [createValueFront("another string value")]),
-      ]);
-    }
-
-    await testUtilsRender(<TestComponent />, { store });
-  }
+  beforeEach(async () => {
+    store = await createTestStore();
+
+    // This is necessary to unblock various event listeners and parsing.
+    await ThreadFront.setSessionId(DEFAULT_SESSION_ID);
+
+    sendSourceToMockEnvironment(
+      createSource({
+        kind: "scriptSource",
+      })
+    );
+
+    sendMessageToMockEnvironment(
+      createConsoleMessage({
+        text: "Plain string message",
+      })
+    );
+    sendMessageToMockEnvironment(
+      createConsoleMessage({
+        argumentValues: [
+          createValue({ value: 123 }),
+          createValue({ value: "string value" }),
+          createValue({ value: true }),
+        ],
+      })
+    );
+    sendMessageToMockEnvironment(
+      createConsoleMessage({
+        text: "Another string message",
+      })
+    );
+    sendMessageToMockEnvironment(
+      createConsoleMessage({
+        argumentValues: [createValue({ value: "another string value" })],
+      })
+    );
+
+    sendLoadedRegionsToMockEnvironment(
+      createLoadedRegions({
+        beginTime: 0,
+        endTime: 1000,
+      })
+    );
```

Comment:
This is still pretty verbose, although I tried to use as many optional params with intelligent defaults as I could.